route/build/src/remap.f90

```diff
@@ -260,49 +260,69 @@ subroutine remap_1D_runoff(runoff_data_in, remap_data_in, basinRunoff, ierr, mes
-  ! case 3: hru in runoff layer is hru polygon identical to river network layer (stored in 1-dimension array)
-  subroutine sort_runoff(runoff_data_in, basinRunoff, ierr, message)
+  ! case 3: sort a given input flux and its ID based on a second given ID array,
+  ! hru in runoff layer is hru polygon identical to river network layer (stored in 1-dimension array)
+  subroutine sort_flux  (ID_in,            & ! input: the array of ids for HRU/seg (can be from 1 to nHRU or nSeg)
+                         IX_in,            & ! input: the array of location of ids for HRU/seg (the same size as ID_in)
+                         flux_in,          & ! input: the array of input fluxes, should be the same size as ID_in
+                         ID_total_in,      & ! input: the sorted array of ids for HRU/seg (should be with nHRU or nSeg) just for check the size
+                         basin_flux_sort,  & ! output: sorted input states and fluxes based on ID_total_in (the same size as ID_total_in)
+                         ierr, message)
   implicit none
   ! input
-  type(runoff)         , intent(in)  :: runoff_data_in   ! runoff for one time step for all HRUs
+  integer(i4b)         , intent(in)    :: ID_in(:)            ! input: the array of ids for HRU/seg (can be from 1 to nHRU or nSeg)
+  integer(i4b)         , intent(in)    :: IX_in(:)            ! input: the array of location of ids for HRU/seg in ID_total_in (can be from 1 to nHRU or nSeg)
+  real(dp)             , intent(in)    :: flux_in(:)          ! input: the array of input fluxes, should be the same size as ID_in
+  integer(i4b)         , intent(in)    :: ID_total_in(:)      ! input: the sorted array of ids for HRU/seg (should be with nHRU or nSeg)
+  ! input/output
+  real(dp)             , intent(inout) :: basin_flux_sort(:)  ! inout: sorted input states and fluxes based on ID_total_in (the same size as ID_total_in)
   ! output
-  real(dp)             , intent(out) :: basinRunoff(:)   ! basin runoff
-  integer(i4b)         , intent(out) :: ierr             ! error code
-  character(len=strLen), intent(out) :: message          ! error message
+  integer(i4b)         , intent(out)   :: ierr             ! error code
+  character(len=strLen), intent(out)   :: message          ! error message
   ! local
-  integer(i4b)                       :: iHRU,jHRU        ! index of basin in the routing layer
-  real(dp)    , parameter            :: xTol=1.e-6_dp    ! tolerance to avoid divide by zero
-  integer(i4b), parameter            :: ixCheck=-huge(iHRU) ! basin to check
+  integer(i4b)                         :: i,j              ! index of basin in the routing layer
 
-  ierr=0; message="sort_runoff/"
+  ierr=0; message="sort_flux/"
 
-  ! loop through hrus in the runoff layer
-  do iHRU=1,size(runoff_data_in%hru_ix)
+!  ! allocate basin_flux_sort is not allocated
+!  if (.not.allocated(basin_flux_sort)) then
```

Comment:
I don't know why it complains, about this line of code, for now commend
```The argument of the ALLOCATED inquiry intrinsic function should have the ALLOCATABLE attribute.   [BASIN_FLUX_SORT]
  if (.not.allocated(basin_flux_sort)) then
---------------------^```

Reply:
This means `basin_flux_sort` is not allocatable array (probably fixed dimension array?)